LoanCalc.java

```diff
@@ -38,17 +47,49 @@ private static double endBalance(double loan, double rate, int n, double payment
 	// the number of periods (n), and epsilon, the approximation's accuracy
 	// Side effect: modifies the class variable iterationCounter.
     public static double bruteForceSolver(double loan, double rate, int n, double epsilon) {
-		// Replace the following statement with your code
-		return 0;
+		double l = loan;
+		double r = rate;
+		int years = n;
+		double e = epsilon;
+		double p = l / n;
+		double sum = endBalance(l, r, years, p);
+		iterationCounter = 0;
+
+		while (sum > 0) {
+			p += e;
+			sum = endBalance(l, r, years, p);
+			iterationCounter ++;
+		}
+
+		return p;
     }
     
     // Uses bisection search to compute an approximation of the periodical payment 
 	// that will bring the ending balance of a loan close to 0.
 	// Given: the sum of the loan, the periodical interest rate (as a percentage),
 	// the number of periods (n), and epsilon, the approximation's accuracy
 	// Side effect: modifies the class variable iterationCounter.
-    public static double bisectionSolver(double loan, double rate, int n, double epsilon) {  
-        // Replace the following statement with your code
-		return 0;
+    public static double bisectionSolver(double loan, double rate, int n, double epsilon) { 
+		double ln = loan;
```

Comment:
If you dont indent to change the values don't need to put them as local variables
I will make this clarification.